File: projects/order-creation/core/src/main/java/com/robinfood/core/dtos/transactionrequestdto/PortionProductDTO.java
package com.robinfood.core.dtos.transactionrequestdto;

import java.io.Serializable;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PortionProductDTO implements Serializable {

    private static final long serialVersionUID = -4019808470258773761L;

    @NotNull
    @Positive
    private Long id;

    @NotBlank
    private String name;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/transactionrequest/PortionProductDTO.kt
package com.robinfood.core.dtos.transactionrequest

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Positive

data class PortionProductDTO (

    @NotNull
    @Positive
    val id: Long,

    @NotBlank
    val name: String
)
